src_c/rect.c

```diff
@@ -695,6 +695,141 @@ pg_rect_collidelistall(pgRectObject *self, PyObject *args)
     return ret;
 }
 
+static SDL_Rect *
+pgRect_FromObjectAndKeyFunc(PyObject *obj, PyObject *keyfunc)
+{
+    SDL_Rect temp;
+    if (keyfunc == NULL) {
+        return pgRect_FromObject(obj, &temp);
+    }
+
+    PyObject *obj_with_rect = PyObject_CallFunctionObjArgs(keyfunc, obj, NULL);
+    if (obj_with_rect == NULL) {
+        return NULL;
+    }
+
+    return pgRect_FromObject(obj_with_rect, &temp);
```

Comment:
```suggestion
    SDL_Rect *ret = pgRect_FromObject(obj_with_rect, &temp);
    Py_DECREF(obj_with_rect);
    if (!ret) {
        PyErr_SetString(
            PyExc_TypeError, "Key function must return rect or rect-like objects");
    }
    return ret;
```
Similar thing as above review, but with another small fix suggestion, `obj_with_rect` is a "new reference" returned by `PyObject_CallFunctionObjArgs`, so after it's used in the function, it must be `Py_DECREF`d

Reply:
changed